language/vm/vm-runtime/src/code_cache/module_cache.rs

```diff
@@ -144,16 +133,16 @@ impl<'alloc> VMModuleCache<'alloc> {
         &self,
         id: &ModuleId,
         fetcher: &F,
-    ) -> VMResult<Option<&'alloc LoadedModule>> {
+    ) -> VMResult<&'alloc LoadedModule> {
         // Currently it is still possible for a script to invoke a nonsense module id function.
         // However, once we have the verifier that checks the well-formedness of the all the linked
         // module id, we should get rid of that ok_or_else case here.
         if let Some(m) = self.map.get(id) {
-            return Ok(Some(&*m));
+            return Ok(&*m);
         }
         let module = match fetcher.get_module(id) {
             Some(module) => module,
-            None => return Ok(None),
+            None => return Err(VMStatus::new(StatusCode::LINKER_ERROR)),
```

Comment:
Hmm LINKER_ERROR is a bit too generic, right? can we attach some more info about what failed to link? or at least a TODO

Reply:
great on the TODO though I am not sure we want another error, but have not thought hard about it